File: Heaps/maxHeap/Java/Node.java
/**
 * String representation of a Node.
 * This module represents a single Node element
 * of a Heap. Such a node has a value and references
 * to its left and right child and to its parent Node.
 * Furthermore it includes a constructor and getters
 * and setters for all that things and a String representation.
 *
 * @author Thomas Lang
 * @version July 15, 2014
 */
public class Node<T extends Comparable<T>>{

    private T value;
    private Node<T> left;
    private Node<T> right;
    private Node<T> parent;

    /**
     * Constructor for a new Node element.
     * It saves the passed value and 
     * initializes all references to null.
     *
     * @param  value   the value to store.
     */
    public Node( T value ){
        this.value = value;
        left       = null;
        right      = null;
        parent     = null;
    }

    /**
     * Getter for the value.
     *
     * @return The value saved in a Node.
     */
    public T getValue(){        return value;  }

    /**
     * Getter for the left child.
     *
     * @return A pointer to the left child of the calling Node.
     */
    public Node<T> getLeft(){   return left;   }

    /**
     * Getter for the right child.
     *
     * @return A pointer to the right child of the calling Node.
     */
    public Node<T> getRight(){  return right;  }

    /**
     * Getter for the parent child.
     *
     * @return A pointer to the parent Node of the calling Node.
     */
    public Node<T> getParent(){ return parent; }

    /**
     * Setter for the Value
     *
     * @param  value   New value to store.
     */
    public void setValue( T value ){       this.value  = value; }

    /**
     * Setter for the left child.
     *
     * @param  node    New <code>left</code> reference.
     */
    public void setLeft(   Node<T> node ){ this.left   = node;  }

    /**
     * Setter for the right child.
     *
     * @param  node    New <code>right</code> reference.
     */
    public void setRight(  Node<T> node ){ this.right  = node;  }

    /**
     * Setter for the parent Node.
     *
     * @param  node    New <code>parent</code> reference.
     */
    public void setParent( Node<T> node ){ this.parent = node;  }

    /**
     * String representation of a single Node.
     *
     * @return String representation of the calling Node.
     */
    public String toString(){  return ( "{" + value + "}" ); }
}

File: Heaps/maxHeap/Python/Node.py

class Node:
    """Class representing a single Node.
       A Node saves a value and references
       to its left and right child and to 
       its parent node."""
    
    def __init__( self, value ):
        """Creates a new Node.
           It saves the passed value and
           initializes the references to
           children and parent Node to None.
           
        @param  vale  The value to save."""
        self.value  = value
        self.left   = None
        self.right  = None
        self.parent = None

    @property
    def value( self ):
        """Returns the saved value in a Node.
        
        @return The saved value."""
        return self.value

    @property
    def left( self ):
        """Returns the reference pointing to 
           the left child of the calling Node.
           
        @return The left pointer."""
        return self.left

    @property
    def right( self ):
        """Returns the reference pointing to
           the right child of the calling Node.
           
        @return The right pointer."""
        return self.right

    @property
    def parent( self ):
        """Returns the reference pointing to 
           the parent Node of the calling Node.
           
        @return The parent pointer."""
        return self.parent

    def setValue( self, value ):
        """Sets the saved value to the passed one.
        
        @param  value  The new value to save."""
        self.value = value

    def setLeft( self, node ):
        """Sets the left reference to the passed one.
        
        @param  node  The new left reference."""
        self.left = node

    def setRight( self, node ):
        """Sets the right reference to the passed one.
        
        @param  node  The new right reference."""
        self.right = node

    def setParent( self, node ):
        """Sets the right reference to the passed one.
        
        @param  node  The new parent reference."""
        self.parent = node

    def __str__( self ):
        """Returns an appropriate String representation
           of the calling node.
           
        @return A String representation of the Node."""
        return "{" + str( self.value ) + "}"
